### PR TITLE
Add Tank_xxy into units

### PR DESCRIPTION
**Pull request by Megumi-X**

### CODE REVIEW

**Comment by Megumi-X** [comment]
加入我自己设计的tank_xxy，将坦克的模型改成了梯形可以分辨前后，窄的一边为前面，宽的一边为后面

File: CMakeSettings.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      "configurationType": "Release",
+      "inheritEnvironments": [ "msvc_x64_x64" ],
+      "buildRoot": "${projectDir}\\out\\build\\${name}",
+      "installRoot": "${projectDir}\\out\\install\\${name}",
+      "cmakeCommandArgs": "",
+      "buildCommandArgs": "",
+      "ctestCommandArgs": ""
+    }
+  ]
+}
```
Comment (LazyJazz) [inline CMakeSettings.json:15]: Delete this

File: src/battle_game/core/units/tank_xxy.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ protected:
+  void TankMove(float move_speed, float rotate_angular_speed);
+  void TurretRotate();
+  void Fire();
+  [[nodiscard]] const char *UnitName() const override;
+  [[nodiscard]] const char *Author() const override;
+
+  float turret_rotation_{0.0f};
+  uint32_t fire_count_down_{0};
+};
```
Comment (LazyJazz) [inline src/battle_game/core/units/tank_xxy.h:21]: You can inherits from unit::Tank directly if you are noting applying any changes

**Review by LazyJazz — CHANGES_REQUESTED** [review]

File: src/battle_game/core/units/tank_xxy.cpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#include "battle_game/core/game_core.h"
+#include "battle_game/graphics/graphics.h"
+#include "tiny_tank.h"
+
+namespace battle_game::unit {
+
+namespace {
+uint32_t tank_body_model_index = 0xffffffffu;
+uint32_t tank_turret_model_index = 0xffffffffu;
+}  // namespace
```
Comment (LazyJazz) [inline src/battle_game/core/units/tank_xxy.cpp:11]: Remove those

**Review by LazyJazz — CHANGES_REQUESTED** [review]

**Review by LazyJazz — APPROVED** [review]
LGTM!

**Pull request merged**